### ISSUE TITLE
remove `_timestamp` properties

### ISSUE THREAD

**Issue by gvwilson** [plotly/dash#3055]
The `_timestamp` properties have been essentially useless since `callback_context` was introduced - they should be removed. cc @alexcjohnson @AnnMarieW

**Comment by AnnMarieW** [comment]
I agree - but do you think it's still important for other languages?  Not sure if that's important if they are no longer being supported.

Here is an [R example](https://github.com/plotly/dash/blob/0d9cd2c2a611e1b8cce21d1d46b69234c20bdb11/components/dash-core-components/vignettes/dash-core-components.Rmd#L555)

Also is it necessary in the [DataTable?](https://github.com/plotly/dash/blob/0d9cd2c2a611e1b8cce21d1d46b69234c20bdb11/components/dash-table/src/dash-table/components/Table/props.ts#L337)

**Comment by alexcjohnson** [comment]
IMO since this is a major version bump of Dash, if other languages want to upgrade their dependencies on the main components it’s reasonable to ask them to support the `callback_context` pattern as well.

**Comment by alexcjohnson** [comment]
Yes, I think we can remove this from `DataTable`, also html:

https://github.com/plotly/dash/blob/0d9cd2c2a611e1b8cce21d1d46b69234c20bdb11/components/dash-html-components/scripts/generate-components.js#L140

I have a vague recollection of there being a reason to keep it for `Store`, but if we don’t have an example of this then probably that too is no longer applicable.

https://github.com/plotly/dash/blob/0d9cd2c2a611e1b8cce21d1d46b69234c20bdb11/components/dash-core-components/src/components/Store.react.js#L219

**Comment by gvwilson** [comment]
Thanks @alexcjohnson and @AnnMarieW - @T4rk1n, please pull it out. (I love deleting code in the morning...)

**Comment by T4rk1n** [comment]
> I have a vague recollection of there being a reason to keep it for Store

I think that is what we have that can trigger a callback without an input on the data itself. Like you may want to update some last saved element but don't need the data to be transfered.

**Comment by gvwilson** [comment]
@T4rk1n @alexcjohnson @AnnMarieW please thumbs-up if you believe we can remove `_timestamp` properties in the upcoming release.

**Comment by AnnMarieW** [comment]
Yes - all except `Store` per comments above

**Comment by alexcjohnson** [comment]
Does that same argument apply to DataTable data_timestamp? Otherwise the props are all small so no reason not to fully transfer them, seems like this is just a performance hack for very large props

**Comment by alexcjohnson** [comment]
TBH though I’m not sure that’s a good enough reason to keep these props around. It’s a very niche case (how often do you really care that something changed but not care what the new value is?), and the same could be accomplished with a small clientside callback bound to the full prop outputting to another prop (an extra store just for this purpose, for example) then you use that as the input to your serverside callback that doesn’t need the full data transferred.

**Comment by AnnMarieW** [comment]
closed as not planned per comment https://github.com/plotly/dash/issues/3148#issuecomment-2640012901
